app/code/Magento/Catalog/Controller/Adminhtml/Product/Gallery/Upload.php

```diff
@@ -98,16 +98,20 @@ public function execute()
                 $mediaDirectory->getAbsolutePath($this->productMediaConfig->getBaseTmpMediaPath())
             );
 
-            $this->_eventManager->dispatch(
-                'catalog_product_gallery_upload_image_after',
-                ['result' => $result, 'action' => $this]
-            );
+            if (is_array($result)) {
+                $this->_eventManager->dispatch(
```

Comment:
That should be called even if `result` is `false`
Please move it on top of `if` statement